go.sum

```diff
@@ -1145,6 +1129,42 @@ honnef.co/go/tools v0.0.1-2020.1.3/go.mod h1:X/FiERA/W4tHapMX5mGpAtMSVEeEUOyHaw9
 lukechampine.com/blake3 v1.1.7/go.mod h1:tkKEOtDkNtklkXtLNEOGNq5tcV90tJiA1vAA12R78LA=
+modernc.org/b v1.0.3 h1:i4Iye3D8KsPozbQndZzccQ/PlIVpMe3U/ET4dhAkGjs=
+modernc.org/b v1.0.3/go.mod h1:bauv72WoSgpBPpK6rQRqE+xVQHzszHS92P3mKx1l3g4=
+modernc.org/db v1.0.4 h1:EbCNg3ajWVZqQL8oEruFYuscbcnfxzgCIURyq7xFhPc=
+modernc.org/db v1.0.4/go.mod h1:h14AWQVee2g0PDe7v6yO7zj6+SpBYRd4+X0oJnaK5LU=
+modernc.org/file v1.0.3/go.mod h1:CNj/pwOfCtCbqiHcXDUlHBB2vWrzdaDCWdcnjtS1+XY=
+modernc.org/file v1.0.4 h1:H6MK8e6Z8b4jeTBE2Ckaf/5DJm88lyEQcwGKObRMRXA=
+modernc.org/file v1.0.4/go.mod h1:f7r4AdK84xOlQjKqHPGv8oJkV9DUm5lv+r/g1xxdevI=
+modernc.org/fileutil v1.0.0/go.mod h1:JHsWpkrk/CnVV1H/eGlFf85BEpfkrp56ro8nojIq9Q8=
+modernc.org/fileutil v1.1.0/go.mod h1:SErFOYfL50xApg0X5XfUOQiku7cTgQ6/8XNvJHKy9eI=
+modernc.org/fileutil v1.1.1 h1:L2sQJIacLy07ufTTA8tEs8MYBSGDZNSgTMv9/uvsrpM=
+modernc.org/fileutil v1.1.1/go.mod h1:HdjlliqRHrMAI4nVOvvpYVzVgvRSK7WnoCiG0GUWJNo=
+modernc.org/golex v1.0.1 h1:EYKY1a3wStt0RzHaH8mdSRNg78Ub0OHxYfCRWw35YtM=
+modernc.org/golex v1.0.1/go.mod h1:QCA53QtsT1NdGkaZZkF5ezFwk4IXh4BGNafAARTC254=
+modernc.org/internal v1.0.2/go.mod h1:bycJAcev709ZU/47nil584PeBD+kbu8nv61ozeMso9E=
+modernc.org/internal v1.0.3/go.mod h1:dvHFQEGEd33HZar0OdSYIm6yen/77eukCqffWSAwQUc=
+modernc.org/internal v1.0.4/go.mod h1:dvHFQEGEd33HZar0OdSYIm6yen/77eukCqffWSAwQUc=
+modernc.org/internal v1.0.5 h1:liZJd7ANjmsxjPK1+Vra6KS8QdqjtypQfjFEnLJydfo=
+modernc.org/internal v1.0.5/go.mod h1:lbE47ueQuXAscba+1ykHkFFi1M3On4ZOreb9LRZ6k4k=
+modernc.org/lex v1.0.0/go.mod h1:G6rxMTy3cH2iA0iXL/HRRv4Znu8MK4higxph/lE7ypk=
+modernc.org/lexer v1.0.0/go.mod h1:F/Dld0YKYdZCLQ7bD0USbWL4YKCyTDRDHiDTOs0q0vk=
+modernc.org/lldb v1.0.4 h1:FWlxVwxUKjuyu1fz3rmXJb4plvsw5F83PYN72ulzFIQ=
+modernc.org/lldb v1.0.4/go.mod h1:AKDI6wUJk7iJS8nRX54St8rq9wUIi3o5YGN3rlejR5o=
+modernc.org/mathutil v1.0.0/go.mod h1:wU0vUrJsVWBZ4P6e7xtFJEhFSNsfRLJ8H458uRjg03k=
+modernc.org/mathutil v1.2.2/go.mod h1:mZW8CKdRPY1v87qxC/wUdX5O1qDzXMP5TH3wjfpga6E=
+modernc.org/mathutil v1.4.1/go.mod h1:mZW8CKdRPY1v87qxC/wUdX5O1qDzXMP5TH3wjfpga6E=
+modernc.org/mathutil v1.5.0 h1:rV0Ko/6SfM+8G+yKiyI830l3Wuz1zRutdslNoQ0kfiQ=
+modernc.org/mathutil v1.5.0/go.mod h1:mZW8CKdRPY1v87qxC/wUdX5O1qDzXMP5TH3wjfpga6E=
+modernc.org/ql v1.4.3 h1:2wTIIo37qVIHCt9OzWy/25YiF3hIqwMwK2iY7vya2Y0=
+modernc.org/ql v1.4.3/go.mod h1:cbkcPWMDiPqu4ruYviVJEnSYNB0AabAv0OEcZ3FZNNk=
+modernc.org/sortutil v1.1.0 h1:oP3U4uM+NT/qBQcbg/K2iqAX0Nx7B1b6YZtq3Gk/PjM=
+modernc.org/sortutil v1.1.0/go.mod h1:ZyL98OQHJgH9IEfN71VsamvJgrtRX9Dj2gX+vH86L1k=
+modernc.org/strutil v1.1.1/go.mod h1:DE+MQQ/hjKBZS2zNInV5hhcipt5rLPWkmpbGeW5mmdw=
+modernc.org/strutil v1.1.3 h1:fNMm+oJklMGYfU9Ylcywl0CO5O6nTfaowNsh2wpPjzY=
+modernc.org/strutil v1.1.3/go.mod h1:MEHNA7PdEnEwLvspRMtWTNnp2nnyvMfkimT1NKNAGbw=
+modernc.org/zappy v1.0.5 h1:XEh6U/ITG9I5Fgl9mBczbaOU7khNcS2+jPVaYlalif4=
+modernc.org/zappy v1.0.5/go.mod h1:Q5T4ra3/JJNORGK16oe8rRAti7kWtRW4Z93fzin2gBc=
```

Comment:
I'm still not too happy to have to add all of these dependencies. Can you remind me why they are needed? Is it for `Darwin` in order to handle migration to the new DB?
If so, could we maybe extract the migration code into it's own self-contained CLI tool?

Reply:
That is a good question. This is the required dependencies for Darwin. We should run `go mod why` on some of these to see why they are needed. 

```
module github.com/ardanlabs/darwin

go 1.16

require (
	github.com/DATA-DOG/go-sqlmock v1.5.0
	modernc.org/ql v1.4.3
)
```

We could separate the migration to a separate CLI tool, but the drawback is the migration is disconnected from the code that needs it. Projects do this for sure, it's just an engineering tradeoff. I don't like it when it can be avoided.

Since you are not vendoring, the module system is designed so this is not an issue. I can't imaging what most of the projects inside Google look like for this.

Reply:
As a solution to this, we've addressed the long(er) dependency list by removing support from Darwin and bumping it to version 2. This should make the list of new dependencies marked in `go.sum` a lot smaller, even if they are not used directly.